agentspeak/__init__.py

```diff
@@ -837,9 +834,7 @@ def unify_annotated(left, right, scope, stack):
         reroll(scope, stack, choicepoint)
 
 
-def unifies_annotated(left, right):
-    scope = {}
-    stack = collections.deque()
+def unifies_annotated(left, right, scope = {}, stack = collections.deque()):
```

Comment:
mutable default arguments are dangerous. there is only one instance of them, and a second call keep using the same dictionary/deque.